samples/replicatedentity-counter/src/main/java/com/example/domain/CounterImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.example.domain;
+
+import com.akkaserverless.javasdk.Reply;
+import com.akkaserverless.javasdk.replicatedentity.*;
+import com.example.CounterApi;
+import com.google.protobuf.Empty;
+
+// tag::class[]
+/**
+ * A Counter represented as a replicated entity.
+ */
+@ReplicatedEntity // <1>
+public class CounterImpl extends CounterInterface {
+    @SuppressWarnings("unused")
+    private final String entityId;
+    private final PNCounter counter;
+
+    public CounterImpl(ReplicatedEntityCreationContext context) { // <2>
+        this.entityId = context.entityId();
+        this.counter = context.state(PNCounter.class).orElseGet(context::newPNCounter);
```

Comment:
By mistake I first used GCounter here, strange that it worked with decrement. Anyway, changed to PNCounter